homeassistant/components/bang_olufsen/media_player.py

```diff
@@ -566,7 +568,42 @@ async def async_play_media(
             if media_id.endswith(".m3u"):
                 media_id = media_id.replace(".m3u", "")
 
-        if media_type in (MediaType.URL, MediaType.MUSIC):
+        if announce:
+            extra = kwargs.get(ATTR_MEDIA_EXTRA, {})
+
+            absolute_volume = extra.get("overlay_absolute_volume", None)
+            offset_volume = extra.get("overlay_offset_volume", None)
+            tts_language = extra.get("overlay_tts_language", "en-us")
+
+            # Construct request
+            overlay_play_request = OverlayPlayRequest()
+
+            # Define volume level
+            if absolute_volume:
+                overlay_play_request.volume_absolute = absolute_volume
+
+            elif offset_volume:
+                # Ensure that the volume is not above 100
+                if not self._volume.level or not self._volume.level.level:
+                    _LOGGER.warning("Error setting volume")
```

Comment:
Instead of logging, should we fail the service by raising `ServiceValidationError`?

Reply:
I think the approach is to just go ahead with the service, even if the absolute volume can't be determined. The overlay will then play at the device's current volume level.

self._volume should pretty much always be known and would only really be unknown if the device is not behaving as expected, so this check is probably not necessary.